sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlanVisitor.scala

```diff
@@ -47,6 +49,16 @@ trait LogicalPlanVisitor[T] {
 
   def default(p: LogicalPlan): T
 
+  def visitSubqueryExpression(p: LogicalPlan): LogicalPlan = {
+    p.transformExpressionsDown {
+      case subqueryExpression: SubqueryExpression =>
+        // trigger subquery's child plan stats propagation
```

Comment:
This is weird. Doesn't EXPLAIN trigger the plan stats propagation?

Reply:
> This is weird. Doesn't EXPLAIN trigger the plan stats propagation?

Yea, EXPLAIN trigger plan stats before build string:
```
  private def stringWithStats(maxFields: Int, append: String => Unit): Unit = {
    val maxFields = SQLConf.get.maxToStringFields

    // trigger to compute stats for logical plans
    try {
      optimizedPlan.stats
    } catch {
      case e: AnalysisException => append(e.toString + "\n")
    }
    // only show optimized logical plan and physical plan
    append("== Optimized Logical Plan ==\n")
    QueryPlan.append(optimizedPlan, append, verbose = true, addSuffix = true, maxFields)
    append("\n== Physical Plan ==\n")
    QueryPlan.append(executedPlan, append, verbose = true, addSuffix = false, maxFields)
    append("\n")
  }
```

Have tested that in subquery, all behavior about statistic is right since when it use statistcs,it will call `plan.stats`.
We do it here just trigger it earlier.

Reply:
If it looks too weird, we can move this to `stringWithStats`

Reply:
Yea let's move it to `stringWithStats`

Reply:
> Yea let's move it to `stringWithStats`

Done